### PR TITLE
FEC-11839 Fire AdError with AD_WATERFALLING_FAILED event for AdLayout

### PR DESCRIPTION
**Pull request by GouravSna**

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt [hunk 1]
```diff
@@ -596,22 +596,34 @@ class PKAdvertisingController: PKAdvertising, IMAEventsListener {
...
+    /**
+     * Handle Error event for AdvertisingConfig
+     * Send AD_WATERFALLING, AD_WATERFALLING_FAILED and ERROR event to the client apps.
+     */
     private fun handleErrorEvent(isAllAdsFailed: Boolean?, adBreakConfig: AdBreakConfig?, error: AdEvent.Error?) {
         log.e("isAdWaterFallingOccurred $hasWaterFallingAds")
         isAllAdsFailed?.let {
             if (hasWaterFallingAds) {
                 if (it) {
+                    log.e("Firing AdWaterFallingFailed event")
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt]: why 2 logs for same blocs?

File: playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt [hunk 2]
```diff
@@ -596,22 +596,34 @@ class PKAdvertisingController: PKAdvertising, IMAEventsListener {
...
                     )
+                    // When the all ads in the waterfalling list fail then with `AdWaterFallingFailed`, we need to fire
+                    // AdEvent.error as well so that Analytics behaves as usual
+                    log.e("Firing AdEvent.error after AdWaterFallingFailed")
+                    error?.let { err ->
+                        messageBus?.post(err)
+                    }
+
                 } else {
+                    log.d("Firing AdWaterFalling event")
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt]: same

File: playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt (hunk 1 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt]: Fixed it

File: playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt (hunk 2 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/ads/PKAdvertisingController.kt]: I think this log is required.

**Review by giladna — APPROVED** [review]

**Pull request merged**